src/test/java/com/googleinterns/zoomtube/mocks/MockRequest.java

```diff
@@ -0,0 +1,45 @@
+package com.googleinterns.zoomtube.mocks;
+
+import static org.mockito.Mockito.mock;
+
+import java.io.BufferedReader;
+import java.io.Reader;
+import java.io.StringReader;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+
+/**
+ * This mocks {@link javax.servlet.http.HttpServletRequest} to allow for testing
+ * servlets. As more tests and servlets are added, you may need to override additional
+ * methods.
+ */
+public class MockRequest extends HttpServletRequestWrapper {
```

Comment:
This is more of a fake than a mock at this point and passing in the mock servlet request in the wrapper is redundant as you are implementing the other methods now..

I noticed that HttpServletRequest is an interface. You should be able to make this a fake without Mockito then.
```public class MockRequest implements HttpServletRequest``` will do this for you. And then people can use your fake requests every where and implement methods as needed.

Reply:
The problem is that the interface has dozens of methods which all need to be implemented.  The Wrapper class wraps an existing implemention so you don't need to reimplement them all.  However, you need to provide an underlying instance, and using mock let's me instantiate one.  I could probably use null instead.  I just want to avoid having to write out every method of the interface--there are a lot

Reply:
You can leave the methods we don't need unimplemented and empty and just implement the ones we actually need. 
Sending a mock in the wrapper class could have other unwanted consequences as well as the  class could be calling methods for which you don't have when...then clauses configured for it. Having a mock in a wrapper class you do not own can become unwieldy to debug. 

Seems to me the two options here are to have the when..then.. clauses in the test itself and mock out HttpServletRequest there or continue with the Fake here but implement it properly. 
I understand you would want the utility methods of ```contentAsString()``` so it sounds like you want the proper Fake?